GVFS/GVFS.Service/GVFSMountProcess.Mac.cs

```diff
@@ -34,19 +44,38 @@ public bool Mount(string repoRoot)
             return true;
         }
 
-        public void Dispose()
+        public class MountLauncher
         {
-        }
+            private ITracer tracer;
 
-        public string GetUserId()
-        {
-            return GVFSPlatform.Instance.GetCurrentUser();
-        }
+            public MountLauncher(ITracer tracer)
+            {
+                this.tracer = tracer;
+            }
 
-        private bool CallGVFSMount(string repoRoot)
-        {
-            InternalVerbParameters mountInternal = new InternalVerbParameters(startedByService: true);
-            throw new NotImplementedException();
+            public virtual bool LaunchProcess(string executablePath, string arguments, string workingDirectory)
+            {
+                ProcessStartInfo processInfo = new ProcessStartInfo(executablePath);
+                processInfo.Arguments = arguments;
+                processInfo.WindowStyle = ProcessWindowStyle.Hidden;
+                processInfo.WorkingDirectory = workingDirectory;
+                processInfo.UseShellExecute = false;
+                processInfo.RedirectStandardOutput = true;
+
+                ProcessResult result = ProcessHelper.Run(processInfo);
+                if (result.ExitCode != 0)
+                {
+                    this.tracer.RelatedError($"{nameof(this.LaunchProcess)} ERROR: {executablePath} {arguments}. Exit code: {result.ExitCode}, {result.Errors}");

```

Comment:
Minor: If you switch to using an `EventMetadata` here it's easier to query the data for specific values (e.g. `result.ExitCode`)